snapcraft/parts/plugins/matter_plugin.py

```diff
@@ -0,0 +1,171 @@
+    """The part properties used by the matter plugin."""
+
+    matter_tag: str
+    matter_zap_tag: str
+
+    @classmethod
+    @overrides
+    def unmarshal(cls, data: Dict[str, Any]) -> "MatterPluginProperties":
+        """Populate class attributes from the part specification.
+
+        :param data: A dictionary containing part properties.
+
+        :return: The populated plugin properties data object.
+
+        :raise pydantic.ValidationError: If validation fails.
+        """
+        plugin_data = plugins.extract_plugin_properties(
+            data, plugin_name="matter", required=["matter_tag", "matter_zap_tag"]
+        )
+        return cls(**plugin_data)
+
+
+class MatterPlugin(plugins.Plugin):
+    """A plugin for matter project.
+
+    This plugin uses the common plugin keywords.
+    For more information check the 'plugins' topic.
+
+    Additionally, this plugin uses the following plugin-specific keywords:
+        - matter-tag
+          (str, no default)
+          The matter branch to use for the build.
+        - matter-zap-tag
+          (str, no default)
+          The zap version to use for the build.
+    """
+
+    properties_class = MatterPluginProperties
+
+    def __init__(
+        self,
+        *,
+        properties: plugins.PluginProperties,
+        part_info: infos.PartInfo,
+    ) -> None:
+        super().__init__(properties=properties, part_info=part_info)
+
+        self.matter_dir = part_info.part_build_dir
+        self.snap_arch = os.getenv("SNAP_ARCH")
+
+    @overrides
+    def get_build_packages(self) -> Set[str]:
+        return {
+            "wget",
+            "unzip",
+            "clang",
+            "pkg-config",
+            "git",
+            "cmake",
+            "ninja-build",
+            "unzip",
+            "libssl-dev",
+            "libdbus-1-dev",
+            "libglib2.0-dev",
+            "libavahi-client-dev",
+            "python3-venv",
+            "python3-dev",
+            "python3-pip",
+            "libgirepository1.0-dev",
+            "libcairo2-dev",
+            "libreadline-dev",
+            "generate-ninja",
+        }
+
+    @overrides
+    def get_build_environment(self) -> Dict[str, str]:
+        return {}
+
+    @overrides
+    def get_build_snaps(self) -> Set[str]:
+        return set()
+
+    @overrides
+    def get_build_commands(self) -> List[str]:
+        options = cast(MatterPluginProperties, self._options)
+        commands = []
+
+        if self.snap_arch == "arm64":
+            commands.extend(
+                [
+                    f"wget --no-verbose https://github.com/project-chip/zap/releases/download/{options.matter_zap_tag}/zap-linux-{self.snap_arch}.zip",
+                    f"unzip -o zap-linux-{self.snap_arch}.zip",
+                    "echo 'export ZAP_INSTALL_PATH=$PWD'",
+                ]
+            )
+
+        """Clone Matter repository if not present"""
```

Comment:
Using string literals is preferred for comment blocks. These are used in several other places for single line comments.
```suggestion
        # Clone Matter repository if not present
```

Reply:
Updated. Please note that there are still several places where `"""` is used for single-line comments, just for formatting alignment with other plugins. Let me know if all single-line comments need to be updated to use `#`.